text/0000-official-utility-buildpacks.md

```diff
@@ -0,0 +1,75 @@
+- Start Date: 2021-06-30
+- Author(s): [@jkutner](https://github.com/jkutner)
+- RFC Pull Request: (leave blank)
+- CNB Pull Request: (leave blank)
+- CNB Issue: (leave blank)
+- Supersedes: N/A
+
+# Summary
+[summary]: #summary
+
+This is a proposal to adopt utility buildpacks as officially supported components of the Cloud Native Buildpacks project.
+
+# Definitions
+[definitions]: #definitions
+
+- **utility buildpack** - a buildpack that provides a generic capability that is useful across many vendors, platforms, and apps
+- **officially supported** - supported and maintained by the the Cloud Native Buildpacks project and its members
+
+# Motivation
+[motivation]: #motivation
+
+Many capabilities of lifecycle and pack do not require a native implementation within these components. These capabilities can (and have been) implemented at the user layer as buildpacks (ex. [inline-buildpack](https://registry.buildpacks.io/buildpacks/hone/inline)). In [PR #168](https://github.com/buildpacks/rfcs/pull/168) we assert that supporting these capabilities natively adds complexity to the implementation of lifecycle, which makes it difficult to innovate and improve.
+
+However, defering to the community and vendors to implement these kinds of utility buildpacks has not necessarily had good results. For example, there are three implementations of the [Procfile buildpack](https://registry.buildpacks.io/buildpacks/paketo-buildpacks/procfile). None of these implementations add significant (if any) advantages over the other. Yet buildpack users are force to decide which one to select. This is an unneccessary burden for a capability that could be provided by default.
+
+We seek to reduce complexity in the lifecycle and provide well maintained utility buildpacks for users by supporting them as official components of the Cloud Native Buildpacks project. This also aligns with our "everything's a buildpack" mantra.
+
+# What it is
+[what-it-is]: #what-it-is
+
+Reserve the `cnb` namespace for officially supported buildpacks. Then, on a case by case basis, we adopt utility buildpacks under this
+
+Examples of potential utility buildpacks include:
+
+- `cnb/profile` - Evaluating `.profile` scripts in the app repo during launch
+- `cnb/inline` - Executing [Inline buildpacks](https://github.com/buildpacks/rfcs/blob/main/text/0048-inline-buildpack.md)
+- `cnb/processes` - Setting app specific process types from a configuration file like `project.toml`
+- `cnb/env` - Setting customer environment variables defined in a configuration file like `project.toml`
+- `cnb/purge` - Purging files from the app repo before export (similar to [`.slugignore`](https://devcenter.heroku.com/articles/slug-compiler#ignoring-files-with-slugignore))
+
+# How it Works
+[how-it-works]: #how-it-works
+
+The utility buildpacks will be owned, supported, and maintained by the [Implementation team](https://github.com/buildpacks/community/blob/main/TEAMS.md#implementation-team).
```

Comment:
Should this be the BAT team?

Reply:
I have no strong opinion

Reply:
If these are things that extensions of the spec/lifecycle I can see the implementation team owning them. Thoughts on this @ekcasey @natalieparellano ?

Reply:
I would vote not to have these maintained by the implementation team to be honest. I think BAT makes most sense.

Reply:
@jabrown85 it just seems somewhat of an awkward fit since these buildpacks are more for platform operators b/c we're pulling out functionality from lifecycle afaict.

Reply:
IMHO, you have to look more at the skills/interests of the team to be maintaining these buildpacks. BAT seems to make the most sense in that light given the BAT team has folks already producing buildpacks and those working on libcnb and trying to make the buildpack development experience as smooth as possible. To me their interests/skills line up very well with this task.

Reply:
@jkutner Can we update this to give ownership to the BAT team. It seems like the BAT team wants to maintain these and the Implementation team doesn't, which seems like the most important consideration to me.

Reply:
For the record I'd like to voice it's a bit out of scope for the BAT team especially as we continue to focus on buildpack tooling which I believe this is out of scope. The overlap is that it'd be nice to dogfood buildpacks with our own tooling, but feels a bit of a miss on the charter. I'm not sure there's a perfect subteam for it.